src/ResourceManager/Search/Commands.Management.Search/SearchService/NewSearchServiceQueryKeyCommand.cs

```diff
@@ -0,0 +1,97 @@
+
+using Microsoft.Azure.Commands.Management.Search.Models;
+using Microsoft.Azure.Commands.Management.Search.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.Search.SearchService
+{
+    [Cmdlet(VerbsCommon.New, SearchServiceQueryKeyNounStr, DefaultParameterSetName = ResourceNameParameterSetName, SupportsShouldProcess = true), OutputType(typeof(PSSearchQueryKey))]
+    public class NewSearchServiceQueryKeyCommand : SearchServiceBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = ParentObjectParameterSetName,
+            HelpMessage = InputObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public PSSearchService ParentObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParentResourceIdParameterSetName,
+            HelpMessage = ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceGroupHelpMessage)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           HelpMessage = QueryKeyNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = ForceHelpMessage)]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSetName, StringComparison.InvariantCulture))
+            {
+                ResourceGroupName = ParentObject.ResourceGroupName;
+                ServiceName = ParentObject.Name;
+            }
+            else if (ParameterSetName.Equals(ParentResourceIdParameterSetName, StringComparison.InvariantCulture))
+            {
+                var id = new ResourceIdentifier(ParentResourceId);
+                ResourceGroupName = id.ResourceGroupName;
+                ServiceName = id.ResourceName;
+            }
+
+            ConfirmAction(Force.IsPresent,
```

Comment:
This should be "ShouldProcess" rather than confirmaction - "Force" should only be applied to cmdlets that make unexpected changes (such as deleting all files when you delete a directory), or very destructive actions that you want a user to be forced to confirm.  Please also remove the -Force parameter.

Apply to all cmdlets that currently have Force but don't have either unexpected changes or especially destructive actions.

Reply:
Ok I see. Now following use ShouldProcess without -Force
NewSearchService
SetSearchService
NewSearchServiceQueryKey